Documentation/gettingstarted/egress-gateway.rst

```diff
@@ -0,0 +1,172 @@
+.. only:: not (epub or latex or html)
+
+    WARNING: You are looking at unreleased Cilium documentation.
+    Please use the official rendered version released here:
+    https://docs.cilium.io
+
+.. _egress_gateway:
+
+**********************
+Egress Gateway (beta)
+**********************
+
+Egress gateway provides a way to configure the egress policy. Without
+this feature, pod traffic could only egress from the node it is running
+on and would be masqueraded with the node IP.
+
+This document explains how to configure Cilium to enable egress gateway
+as well as how to configure egress NAT policy to route and SNAT the egress
+traffic for a specific workload.
+
+.. include:: ../beta.rst
+
+Configure Cilium
+================
+
+.. include:: k8s-install-download-release.rst
+
+The Cilium Egress Gateway feature relies on :ref:`kubeproxy-free`,
+follow the guide to create a new deployment. The feature is disabled by default.
+Enable the feature by setting the ``localRedirectPolicy`` value to ``true``.
+
+.. parsed-literal::
+
+   helm install cilium |CHART_RELEASE| \\
+     --set enableEgressGateway=true
+
+Verify that Cilium agent pod is running.
+
+.. code-block:: shell-session
+
+    $ kubectl -n kube-system get pods -l k8s-app=cilium
+    NAME           READY   STATUS    RESTARTS   AGE
+    cilium-5ngzd   1/1     Running   0          3m19s
+
+Validate that the Cilium Egress NAT Policy CRD has been registered.
+
+.. code-block:: shell-session
+
+	   $ kubectl get crds
+	   NAME                              CREATED AT
+	   [...]
+	   ciliumegressnatpolicy.cilium.io   2021-03-26T02:11:21Z
+
+
+Optionally, Egress Gateway support can be enabled by setting
+``enable-egress-gateway`` to ``true`` in ConfigMap. By default, it is disabled.
+
+To set ``enable-egress-gateway`` to ``true``, it requires
+``enable-bpf-masquerade`` and ``kube-proxy-replacement``. It cannot be enabled
+with either of these 2 options set as ``false`` or ``disabled``.
+
+Architecture
+============
+
+Overall, some nodes in the cluster will be assigned as gateway nodes. These gateway nodes not
+only have a node IP, but also have additional pools of egress IPs attached to it. These egress IP
+addresses can be associated with certain workloads (pods) that run in the cluster. When certain
+packets are selected, instead of exiting the cluster immediately, they will be forwarded to a gateway
+node and SNAT-ed with an egress IP there. A tunnel interface will be created when Egress Gateway is
+enabled. The tunnel is for forwarding with Egress Gateway. All other traffic will keep their
+own behavior based on which tunnel mode is selected.
+
+The following 2 diagrams show the packet flow for both egress and return traffic when using direct
+routing mode. Note that the return packet is not going over a tunnel, while forwarding traffic is. 
+
+.. image:: images/cilium_egress_gateway_flow.png
+   :scale: 70%
+
+.. image:: images/cilium_egress_gateway_return_flow.png
+   :scale: 70%
```

Comment:
This section is great! But it's usually something we include in the blog post for the new release (I'll ping you when we start writing that). If we include it in the documentation, then (1) it should probably be somewhere in Concepts instead and (2) that means we'll try to maintain it. I would just remove this and keep it on the side for the blog post.

Reply:
I think it makes sense. Removed for this guide.